packages/next/build/index.ts

```diff
@@ -500,6 +502,8 @@ export default async function build(dir: string, conf = null): Promise<void> {
           } else if (result.static && customAppGetInitialProps === false) {
             staticPages.add(page)
             isStatic = true
+          } else if (result.serverProps) {
```

Comment:
Server props have to be checked before this branch right?

`result.prerender` => `result.serverProps` => `result.static`